### ISSUE TITLE
Warn when echo'ing: accidental globbing with "?", history expansion with "!!"

### ISSUE THREAD

**Issue by thp-hatch** [koalaman/shellcheck#1754]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

echo File checksum mismatch! Input changed?
```

```sh
#!/bin/bash

echo Something did not work!!
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

Warn the user that if a file that matches the "changed?" glob expression (e.g. `changedd` or `changed1`) exists, the output message will change. To prevent that, quote the string.

#### Example error case:

```
$ touch changedd
$ echo File checksum mismatch! Input changed?
File checksum mismatch! Input changedd
```

```
$ ls
...
$ echo foo!!
fools
```

#### Proposed fix suggestion:

Surround with double quotes.

```
echo "File checksum mismatch! Input changed?"
```

```
echo "foo!!"
```

Or alternatively:

```
echo File checksum mismatch! Input "changed?"
```
